src/consensus/role_follower.rs

```diff
@@ -491,6 +512,11 @@ impl Consensus {
             )
             .is_err()
         {
+            // Swap back
+            std::mem::swap(
```

Comment:
when I see that, I ask myself if it wouldn't be cleaner to edit `try_commit_current_proposal` to `try_commit_proposal` and pass the proposal as parameter :thinking: 

could be a future refacto